crates/storage-query-datafusion/src/remote_query_scanner_manager.rs

```diff
@@ -67,36 +65,60 @@ impl Debug for RemoteScannerManager {
 }
 
 pub enum PartitionLocation {
-    Local { scanner: Arc<dyn ScanPartition> },
+    Local,
     Remote { node_id: NodeId },
 }
 
-impl RemoteScannerManager {
-    pub fn new(
-        metadata: Metadata,
-        partition_routing: PartitionRouting,
-        remote_scanner: Arc<dyn RemoteScannerService>,
-    ) -> Self {
-        Self {
-            my_node_id: OnceLock::new(),
-            metadata: Some(metadata),
-            partition_routing,
-            remote_scanner,
-            local_store_scanners: LocalPartitionScannerRegistry::default(),
+pub trait PartitionLocator: Send + Sync + 'static {
+    fn get_partition_target_node(
+        &self,
+        partition_id: PartitionId,
+    ) -> anyhow::Result<PartitionLocation>;
+}
+
+#[derive(Clone)]
+struct MetadataAwarePartitionLocator {
+    partition_routing: PartitionRouting,
+    metadata: Metadata,
+}
+
+pub fn create_partition_locator(
+    partition_routing: PartitionRouting,
+    metadata: Metadata,
+) -> Arc<dyn PartitionLocator> {
+    Arc::new(MetadataAwarePartitionLocator {
+        partition_routing,
+        metadata,
+    })
+}
+
+impl PartitionLocator for MetadataAwarePartitionLocator {
+    fn get_partition_target_node(
+        &self,
+        partition_id: PartitionId,
+    ) -> anyhow::Result<PartitionLocation> {
+        let my_node_id = self.metadata.my_node_id();
+        match self.partition_routing.get_node_by_partition(partition_id) {
+            None => {
+                self.partition_routing.request_refresh();
```

Comment:
I think we can take this out now; partition routing will do this by itself now.

```suggestion
```

Ref: https://github.com/restatedev/restate/blob/206e3ee6df6987fb4cda48040ffbb29a231de4d3/crates/core/src/partitions/mod.rs#L71-L77